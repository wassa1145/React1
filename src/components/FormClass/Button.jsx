import React, { Component } from 'react';

export class Button extends Component {
  render() {
    return <button disabled={this.props.disabled} className="form__bottom" onClick={this.props.click}>{ this.props.name}</button>
  }
}
