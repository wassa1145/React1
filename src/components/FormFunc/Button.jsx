export const Button = (props) => {
    return <button className="form__bottom" disabled={props.disabled} onClick={props.click}>{props.name}</button>
}
