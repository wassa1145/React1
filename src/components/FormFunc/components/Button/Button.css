.form__button {
  background-color: #fff !important;
  height: 56px;
  align-self: flex-end;  
}
