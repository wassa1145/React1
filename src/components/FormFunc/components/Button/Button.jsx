import React from 'react';
import './Button.css';

export const Button = (props) => {
  return (
    <button
      className="form__button"
      disabled={props.disabled}
      data-testid="submit-button"
      type="submit"
    >
      {props.name}
    </button>
  );
};
