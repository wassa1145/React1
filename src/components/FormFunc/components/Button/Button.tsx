import React, { FC } from 'react';
import { Button as ButtonUI } from '@mui/material';
import './Button.css';

interface ButtonProps {
  disabled: boolean;
  name: string;
}

export const Button: FC<ButtonProps> = (props) => {
  return (
    <ButtonUI
      variant="outlined"
      className="form__button"
      disabled={props.disabled}
      data-testid="submit-button"
      type="submit"
    >
      {props.name}
    </ButtonUI>
  );
};
