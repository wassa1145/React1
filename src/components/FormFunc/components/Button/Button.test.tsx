import React from 'react';
import { Button } from './Button';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('Button', () => {
  it('render component', () => {
    render(<Button name="Отправить"/>);
  });

  it('render with snapshot', () => {
    const { asFragment } = render(<Button name="Отправить"/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('render component with text', () => {
    render(<Button name="Отправить" />);
    expect(screen.getByText(/Отправить/)).toBeInTheDocument();
  });

  it('button is disabled', () => {
    render(<Button name="Отправить" disabled />);
    expect(screen.getByTestId('submit-button')).toBeDisabled();
  });
});
