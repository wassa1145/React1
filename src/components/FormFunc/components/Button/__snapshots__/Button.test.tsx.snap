// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render with snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root form__button css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
    data-testid="submit-button"
    tabindex="0"
    type="submit"
  >
    Отправить
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
